src/sage/rings/species.py

```diff
@@ -0,0 +1,406 @@
+                and self._C.degree() <= other._C.degree()
+                and (self._C.degree() < other._C.degree() or
+                     (GAP_FAIL != libgap.ContainedConjugates(self.subgroup_of(),
+                                                             other._C, self._C, True))))
+
+    def __eq__(self, other):
+        r"""
+        Return if this element is equal to ``other``.
+
+        Two elements compare equal if they are conjugate subgroups in the parent group.
+
+        TESTS::
+
+            sage: G = SymmetricGroup(4)
+            sage: B = BurnsideRing(G)
+            sage: H1 = PermutationGroup([(1,2)])
+            sage: H2 = PermutationGroup([(2,3)])
+            sage: B[H1] == B[H2]
+            True
+        """
+        return (isinstance(other, ConjugacyClassOfSubgroups)
+                and self._C.degree() == other._C.degree() and
+                _is_conjugate(self.subgroup_of(), self._C, other._C))
+
+class AtomicSpecies(ConjugacyClassOfDirectlyIndecomposableSubgroups):
+    def __init__(self, parent, C, multicardinality):
+        r"""
+        An atomic species centered on a given multicardinality: a directly
+        indecomposable subgroup paired with an integer vector.
+        """
+        total_cardinality = sum(multicardinality)
+        if C.degree() != sum(multicardinality):
+            raise ValueError(f"Degree of {C} (= {C.degree()}) must equal the total cardinality {total_cardinality}")
+        G = SymmetricGroup(total_cardinality).young_subgroup(multicardinality)
+        if not C.is_subgroup(G):
+            raise ValueError(f"{C} is not a subgroup of {G}")
+        ConjugacyClassOfDirectlyIndecomposableSubgroups.__init__(self, parent, C)
+        self._mc = multicardinality
+        self._tc = total_cardinality
+
+    @cached_method
+    def subgroup_of(self):
+        r"""
+        Return the group which the underlying permutation group
+        of this atomic species belongs to.
+        """
+        return SymmetricGroup(self._tc).young_subgroup(self._mc)
+
+    def __hash__(self):
+        r"""
+        Return the hash of the atomic species.
+        """
+        return hash(tuple([self._C, self._mc, self._tc]))
+
+    @cached_method
+    def _element_key(self):
+        r"""
+        Return a lookup key for ``self``.
+        """
+        return tuple([*super()._element_key(), self._mc])
+
+    def __eq__(self, other):
+        r"""
+        Two atomic species are equal if the underlying groups are conjugate,
+        and their multicardinalities are equal.
+        """
+        if parent(self) != parent(other):
+            return False
+        return self._mc == other._mc and super().__eq__(other)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+        """
+        return "{" + f"{self._C.gens_small()}: {self._mc}" + "}"
+
+class MultivariateAtomicSpecies(UniqueRepresentation, Parent, ElementCache):
```

Comment:
This should go away.

Reply:
This is the parent class for atomic species (I'll rename it to `AtomicSpecies` and `AtomicSpecies` to `AtomicSpeciesElement`). Why remove it?